library/src/main/java/com/danikula/videocache/HttpProxyCacheServer.java

```diff
@@ -114,20 +122,35 @@ private void makeSureServerWorks() {
 
     private boolean pingServer() throws ProxyCacheException {
         String pingUrl = appendToProxyUrl(PING_REQUEST);
-        HttpUrlSource source = new HttpUrlSource(pingUrl);
+
+        StringBuilder sb = new StringBuilder();
+        InputStream is = null;
+
         try {
-            byte[] expectedResponse = PING_RESPONSE.getBytes();
-            source.open(0);
-            byte[] response = new byte[expectedResponse.length];
-            source.read(response);
-            boolean pingOk = Arrays.equals(expectedResponse, response);
-            Log.d(LOG_TAG, "Ping response: `" + new String(response) + "`, pinged? " + pingOk);
-            return pingOk;
-        } catch (ProxyCacheException e) {
-            Log.e(LOG_TAG, "Error reading ping response", e);
+            //Don't try to use a proxy to access our proxy. Proxyception.
+            HttpURLConnection connection = (HttpURLConnection) new URL(pingUrl).openConnection(Proxy.NO_PROXY);
```

Comment:
Are you sure this fix resolves issue? Bypassing system proxy allows **only ping cacheproxy**, but MediaPlayer will not whatever.